### ISSUE TITLE
Action request: filter / sort

### ISSUE THREAD

**Issue by tuchk4** [tuchk4/apop#15]
```js
const data = [
 { x: 0, y: 0},
 { x: 1, y: 1},
 { x: 2, y: 2},
];

rmk(
  rmk.filter(state => state.x === 0 && state.y === 0),
)(data) 
```

Same for `sort` action

**Comment by marvelsrp** [comment]
@tuchk4 For array found methods:
data.filter(item =>  item.x === 0 && item.y === 0)
data.sort(item =>  item.x)

**Comment by tuchk4** [comment]
Yes, this is native js array methods.
Just thought that is should be useful to use along with other rmk actions.

```js
rmk(
  // ... other actions
  rmk.filter(state => state.x === 0 && state.y === 0),
)(data) 
```
Instead of:

```js
rmk(
  // ... other actions
)(data).filter(state => state.x === 0 && state.y === 0);

// or

let updated = rmk(
  // ... other actions
)(data);

updated = updated.filter(state => state.x === 0 && state.y === 0);
```

**Comment by tuchk4** [comment]
Suggest:

Current way to use *swap*/ *replace* / *filter* / *sort*:
```
const result = formula(['foo', 'bar', 'baz']);
const formula = rmk(rmk.swap(1, 2));
formula(['foo', 'bar', 'baz']); // expect ['foo', 'baz', 'bar']
```
This wont work brcause actions takes array items. So it is not possible to swap array items or sort or filter it.

Suggest to split into:
- object actions
- array actions

```
const result = formula(['foo', 'bar', 'baz']);
const formula = rmk().swap(1, 2)
formula(['foo', 'bar', 'baz']); // expect ['foo', 'baz', 'bar']
```
There are native array methods + custom rmk actions:
```
const formula = rmk()
  .swap(1, 2)
  .filter(() => {})
  .sort(() => {})
  .replace(1, 2)
```

Note that rmk should throw exception  it call array actions for object.

```
rmk(clear())({

}).filter(() => {}); // throw exception that array actions only for arrays
```

**Comment by tuchk4** [comment]
or maybe for array actions:

```js
const formula = rmk(
  clear(),
  pick('id', 'name')
).array(
  swap(1, 2)
  filter(() => {}),
  sort(() => {}),
  replace(1, 2)
);

formula(data); 

```

**Comment by marvelsrp** [comment]
implemented in [apop 0.0.1](https://github.com/tuchk4/apop/commit/d00a2a3a006997daf7348a0b67f5bfd4c957efd0)
